core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java

```diff
@@ -249,6 +264,8 @@ public boolean invalid() {
     public void handleClientOperation(
             @Transient SqlCompiler compiler,
             @Transient AssociativeCache<Object> factoryCache,
+            @Transient AssociativeCache<Object> portalFactoryCache,
```

Comment:
AssociativeCache is non-deterministic. Entities that are used together (e.g. `portalFactoryCache` and `statementToBindVariableTypeCache`) must be stored together in the same cache. Otherwise it is possible for only one of the caches to lose its value making pair inconsistent.

To that end I guess there is no test that would saturate these caches to breaking point. We need one.